contrib/babelfishpg_tsql/src/pltsql_coerce.c

```diff
@@ -1084,6 +1086,94 @@ tsql_coerce_string_literal_hook(ParseCallbackState *pcbstate, Oid targetTypeId,
 	return NULL;
 }
 
+static bool
+is_tsql_char_type(Oid type)
+{
+	common_utility_plugin *utilptr = common_utility_plugin_ptr;
+	return utilptr->is_tsql_bpchar_datatype(type) ||
+			utilptr->is_tsql_nchar_datatype(type);
+}
+
+static bool
+is_tsql_varchar_type(Oid type)
+{
+	common_utility_plugin *utilptr = common_utility_plugin_ptr;
+	return utilptr->is_tsql_varchar_datatype(type) ||
+			utilptr->is_tsql_nvarchar_datatype(type);
+}
+
+static bool
+expr_is_null_or_unknown(Node *expr)
+{
+	Oid type = exprType(expr);
+	return type == UNKNOWNOID || (
+		IsA(expr, Const) && (type == TEXTOID || ((Const*)expr)->constisnull));
+}
+
+static Oid
```

Comment:
The return of this func
if it's invalidOid, it'll just follow the normal PG select_common_type
We should leave a comment here